resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -198,5 +198,6 @@ private[spark] object KubernetesConf {
       .replaceAll("\\.", "-")
       .replaceAll("[^a-z0-9\\-]", "")
       .replaceAll("-+", "-")
+      .replaceAll("^-", "")
```

Comment:
Why? this character is allowed later in the name. I don't get why you can't just check a-z? I don't think this PR is making progress, and will close it

Reply:
I replaced the first character，ok?

Reply:
I updated the code for checking a-z.
`Character.isLetter(preferredServiceName.charAt(0)`

Reply:
`.replaceAll("^-", "")` ,we can use the code to avoid that the first character of all kubernetes resources(Pod ,Service,Secret,etc) name is not '-'

Reply:
This should be reverted then?